Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIBookmarks.mm

```diff
@@ -895,6 +895,96 @@ void configureMoveBookmarksDelegate(TestWebExtensionManager *manager)
+
+        @"let results1 = await browser.bookmarks.search('dog');",
+        @"browser.test.assertEq(2, results1.length, 'S1: Should find 2 bookmarks for dog');",
+        @"browser.test.assertEq(bm1.id, results1[0].id, 'S1: First result should be bm1');",
+        @"browser.test.assertEq(bm7.id, results1[1].id, 'S1: Second result should be bm7');",
+
+        @"let results2 = await browser.bookmarks.search('dog site');",
+        @"browser.test.assertEq(1, results2.length, 'S2: Should find 1 bookmark for dog site');",
+        @"browser.test.assertEq(bm7.id, results2[0].id, 'S2: Result should be bm7');",
+
+        @"let results3 = await browser.bookmarks.search('Animals');",
+        @"browser.test.assertEq(3, results3.length, 'S3: Should find 1 folder for Animals');",
+        @"browser.test.assertEq(folderA.id, results3[0].id, 'S3: Result should be folderA');",
+        @"browser.test.assertEq(bm1.id, results3[1].id, 'S3: Result should be folderA');",
+        @"browser.test.assertEq(bm2.id, results3[2].id, 'S3: Result should be folderA');",
+
+        @"let results4 = await browser.bookmarks.search('nonexistent');",
+        @"browser.test.assertEq(0, results4.length, 'S4: Should find 0 bookmarks for nonexistent');",
+
+        @"let results5 = await browser.bookmarks.search({ query: 'cat' });",
+        @"browser.test.assertEq(1, results5.length, 'S5: Should find 1 bookmark for query cat');",
+        @"browser.test.assertEq(bm2.id, results5[0].id, 'S5: Result should be bm2');",
+
+        @"let results6 = await browser.bookmarks.search({ title: 'Bird' });",
+        @"browser.test.assertEq(1, results6.length, 'S6: Should find 1 bookmark for title Bird');",
+        @"browser.test.assertEq(bm3.id, results6[0].id, 'S6: Result should be bm3');",
+
+        @"let results7 = await browser.bookmarks.search({ url: 'flowers.com' });",
+        @"browser.test.assertEq(0, results7.length, 'S7: Should find 2 bookmarks for url flowers.com');",
+        @"let results8 = await browser.bookmarks.search('flowers.com');",
+        @"browser.test.assertEq(2, results8.length, 'S8: Should find 2 bookmarks for url flowers.com');",
+        @"browser.test.assertEq(bm4.id, results8[0].id, 'S8: First result should be bm4');",
+        @"browser.test.assertEq(bm5.id, results8[1].id, 'S8: Second result should be bm5');",
+
+        @"let results9 = await browser.bookmarks.search({ title: 'Dog', url: 'http://animals.com/dog' });",
+        @"browser.test.assertEq(1, results9.length, 'S9: Should find 1 bookmark for title Dog and url animals.com');",
+        @"browser.test.assertEq(bm1.id, results9[0].id, 'S9: Result should be bm1');",
+
+        @"let results10 = await browser.bookmarks.search({ query: 'animal', title: 'Wild Animal' });",
+        @"browser.test.assertEq(1, results10.length, 'S10: Should find 1 bookmark for query animal and title Wild Animal');",
+        @"browser.test.assertEq(bm6.id, results10[0].id, 'S10: Result should be bm6');",
```

Comment:
It would be neat to have a test where `query` match the URL and title match the title.

Maybe:
`@"let results10 = await browser.bookmarks.search({ query: 'nature', title: 'Wild Animal' });",`